### DISCUSSION TITLE
We should not assume users want to add a reply

### DISCUSSION THREAD

**Discussion by jasonLaster** [RecordReplay/devtools#4227] [Q&A] [answered]
We currently assume when the user clicks on a comment that they want to leave a reply. Because of this, we open the reply editor and focus in it. 

What I'm finding is that most of the time, i'm clicking the comment with the goal of seeing what the app looked like at that point or right after. This means i often click the comment and then press the space bar. Today, this feels like a bug because i expect the video to play, but instead i'm leaving a comment.

https://app.replay.io/recording/d00dd0d2-94be-4d0c-ba48-6815047b9c4a?point=56466228345455674171628531734285027&time=20573&hasFrames=false#

**Comment by jaril** [comment]
I agree. The click -> focus on the textbox is a little big aggressive. Taking two clicks to navigate to a comment and add a reply is reasonable. 

If we really do want a one-click-to-start-leaving-a-reply experience, we should:
- expose a reply button on **all** comments regardless of being paused, or
- show a reply button on hover.

**Reply by jasonLaster** [reply]
true. the frame.io ui of having a replay button feels good.

**Comment by hbenl** [comment]
Somewhat related: #4229

**Comment by jcmorrow** [comment]
Yeah, I know I've been trying different things out with focus and auto-opening/closing, but I do think in general it's better to lean towards the "dumber" way to do things (buttons, things not appearing/disappearing). I think the place where this breaks down is when a UI item is so general-purpose that you can reasonably come up with a list of 15 things that a user wants to do with an item. In our case, I don't think we have that problem. For me, comments are primarily used to:

1. Navigate through the timeline
2. Edit or delete a comment
3. Add a reply

I think Editing/Deleting happens a lot for me because I QA a lot, most likely people are navigating much of the time, adding replies less often (do we have data on this? Would be awesome to know how many clicks on comments led to replies, how much do people click on their own comments, etc.), and deleting/editing even less often.

I think personally, I'm in favor of having clicking on a card always mean seeking to that point. This is consistent with the console. I think, as @jaril mentioned, adding a reply should be a button, near the bottom where you would expect to type a reply. Personally I would probably lean towards always visible, if a little subtle, but maybe that's just because I'm from the 90's.

**Reply by jasonLaster** [reply]
This feels right to me.

re data: sadly very few people have used comments up to this point. This is a big reason why polishing markers and comments felt so important. We know that collaboration is important and we want to make sure that we're doing a great job.

Here's how frame.io [video](https://app.frame.io/reviews/880814bc-23fc-4202-8014-da2625381d90/e345b37e-0961-4a8d-9004-6a4b9a2171f4). I feel like they're the gold standard. By the way, two advantages of always visible are (a) the ui does not jump, (b) users have a one click experience when they do want to reply.

re being from the 90's. We need some Gen Z'ers quick!

**Comment by jonbell-lot23** [comment, marked as answer]
As of March 7 2022, pretty sure we've addressed this?

**Reply by jcmorrow** [reply]
Yep. https://github.com/RecordReplay/devtools/commit/3f3cccee24499dcc2dd79aa5ec26e5f618fb38da addressed this I believe. I think our dicussions just stay stale a lot longer than our issues 😓

**Reply by jonbell-lot23** [reply]
How do we mark something as an answer? I'm trying to clean up but I can't find that button! :)

**Reply by jcmorrow** [reply]
For me it shows up like this (this says unmark because I already marked it, but you get the point):

![CleanShot 2022-03-07 at 12 31 05](https://user-images.githubusercontent.com/5903784/157113095-203bd9f3-6b20-43de-bdf8-bd84dfe00fb4.png)

I'm not sure what the conditions are for that button showing up, but maybe it can't be a reply? (Or maybe you can't mark your own as answer? I have no idea)
